tests/testMeasureApCorr.py

```diff
@@ -0,0 +1,204 @@
+                                                 numSources=5, dimension=self.exposure.getWidth())
+
+    def tearDown(self):
+        del self.schema
+        del self.meas_apCorr_task
+        del self.exposure
+        del self.sourceCat
+
+    def testAddFields(self):
+        """Instantiating the task should add one field to the schema"""
+        self.assertTrue("apcorr_" + self.name + "_used" in self.schema.getNames())
+
+    def testReturnApCorrMap(self):
+        """The measureApCorr task should return a structure with a single key "apCorrMap"""
+        struct = self.meas_apCorr_task.run(catalog=self.sourceCat, exposure=self.exposure)
+        self.assertEqual(struct.getDict().keys(), ['apCorrMap'])
+
+    def testApCorrMapKeys(self):
+        """An apCorrMap structure should have two keys, based on the name supplied to addApCorrName()"""
+        apfluxName = self.apname + "_flux"
+        apfluxSigmaName = self.apname + "_fluxSigma"
+        struct = self.meas_apCorr_task.run(catalog=self.sourceCat, exposure=self.exposure)
+        key_names = [apfluxName, apfluxSigmaName]
+        self.assertEqual(set(struct.apCorrMap.keys()), set(key_names))
+
+    def testTooFewSources(self):
+        """ If there are too few sources, check that an exception is raised."""
+        apFluxName = self.apname + "_flux"
+        catalog = afwTable.SourceCatalog(self.schema)
+        self.assertRaises(RuntimeError, self.meas_apCorr_task.run, catalog=catalog, exposure=self.exposure)
+
+    def testSourceNotUsed(self):
+        """ Check that a source outside the bounding box is flagged as not used (False)"""
+        fluxName = self.name + "_flux"
+        apCorrFlagKey = self.schema.find("apcorr_" + self.name + "_used").key
+        source = self.sourceCat.addNew()
+        source_test_flux = 5.1
+        source_test_centroid = afwGeom.Point2D(15, 7.1)
+        fluxKey = self.schema.find(fluxName).key
+        centroidKey = afwTable.Point2DKey(self.schema["slot_Centroid"])
+        source.set(fluxKey, source_test_flux)
+        source.set(centroidKey, source_test_centroid)
+        self.meas_apCorr_task.run(catalog=self.sourceCat, exposure=self.exposure)
+        self.assertFalse(self.sourceCat[apCorrFlagKey][-1])
+
+    def testSourceUsed(self):
+        """ Check that valid sources inside the bounding box that are used have their flags set to True"""
+        inputFilterFlagKey = self.schema.find(self.meas_apCorr_task.config.starSelector.field).key
+        self.meas_apCorr_task.run(catalog=self.sourceCat, exposure=self.exposure)
+        self.assertTrue(self.sourceCat[inputFilterFlagKey].all())
+
+    def testApertureMeasOnes(self):
+        """ Check that sources with aperture fluxes exactly the same as their catalog fluxes
+            returns an aperture correction map of 1s"""
+        apFluxName = self.apname + "_flux"
+        sourceCat = apCorrTestSourceCatalog(schema=self.schema, name=self.name, apname=self.apname,
```

Comment:
Given the frequent need to call this, I'm not convinced making a default catalog is an improvement. Instead how about making apCorrTestSourceCatalog a method so that it can have sensible defaults for most of its arguments (in fact many of its argument could be eliminated). Then calling it would typically require only overriding the one or two arguments that actually matter. I think this would make it much easier to understand what is going on.